core/src/main/java/org/apache/accumulo/core/client/sample/AbstractHashSampler.java

```diff
@@ -57,14 +58,28 @@
   private int modulus;
 
   private static final Set<String> VALID_OPTIONS = Set.of("hasher", "modulus");
+  private static final Set<String> VALID_VALUES_HASHER = Set.of("murmur3_32", "md5", "sha1");
 
   /**
    * Subclasses with options should override this method and return true if the option is valid for
    * the subclass or if {@code super.isValidOption(opt)} returns true.
    */
+  @Override
+  public void validateOptions(Map<String,String> config) {
+    for (Map.Entry<String,String> entry : config.entrySet()) {
+      checkArgument(isValid(entry.getKey()), "Unknown option: %s", entry.getKey());
+
+      if (entry.getKey().equals("hasher"))
+        checkArgument(isValid(entry.getValue()), "Unknown value for hasher: %s", entry.getValue());
 
-  protected boolean isValidOption(String option) {
-    return VALID_OPTIONS.contains(option);
+      if (entry.getKey().equals("modulus"))
+        checkArgument(Integer.parseInt(entry.getValue()) > 0,
+            "Improper Integer value for modulus: %s", entry.getValue());
+    }
+  }
+
+  protected boolean isValid(String entry) {
+    return VALID_OPTIONS.contains(entry) || VALID_VALUES_HASHER.contains(entry);
```

Comment:
Each half of this method's implementation is checking for different things. The left side of the `||` checks keys, whereas the right side checks values. Each case should probably just be inlined into their respective areas of the code, and this method can be deleted.

Reply:
Or, since this method is protected (public API), it should be deprecated, and its original implementation left intact.

Reply:
My lasted commit includes these changes. I did revert that function and deprecated it. Let me know if I need to do something else for it or not.